compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -713,6 +713,7 @@ private BType getTypeOfHexFloatingPointLiteral(BLangLiteral literalExpr, Object
     }
 
     public BType setLiteralValueAndGetType(BLangLiteral literalExpr, BType expType, AnalyzerData data) {
+        literalExpr.isFiniteContext = false;
```

Comment:
@KRVPerera can you remember the reason behind this verdict

Reply:
This is by default false. Logic inside this function is the one which set this to true. Original problem is some nodes are getting cloned at desugar had this set to true. If this is set to true it prevents  type checker progressing and setting the result.